### PR TITLE
[1248] Expose course enrichments

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Course enrichment data, this is required for the frontend course details view

### Changes proposed in this pull request
- Optional include on courses endpoint

### Guidance to review
example: `http://localhost:3001/api/v2/providers/2AT/courses/35L8?include=enrichments`

TODO: Specs

### CODE REVIEW

**Comment by dankmitchell** [comment]
@benilovj good point. Let's discuss

**Comment by defong** [comment]
max of 2, latest published, latest draft

File: app/serializers/api/v2/serializable_course_enrichment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module API
+  module V2
+    class SerializableCourseEnrichment < JSONAPI::Serializable::Resource
+      type 'course_enrichments'
+
+      attributes :json_data, :status, :has_been_published_before?
```
Comment (misaka) [inline app/serializers/api/v2/serializable_course_enrichment.rb]: Should we really be exposing this as `json_data`? I think we should be hiding this particular implementation detail and exposing the attributes explicitly.

Comment (misaka) [reply app/serializers/api/v2/serializable_course_enrichment.rb]: Can we add tests to `spec/serializers/api/v2/serializable_course_spec.rb` to ensure that we serialize the expected attributes?

Comment (misaka) [reply app/serializers/api/v2/serializable_course_enrichment.rb]: The more I speak & think about it, it feels like the right thing to do here is to protect `manage-courses-frontend` from the enrichments as much as possible. I think we should explore exposing the course details in a 'published' form and a 'current' form (which would be the latest published or draft version).

**Pull request closed without merging**